scylla-cdc/src/stream_generations.rs

```diff
@@ -0,0 +1,531 @@
+) WITH CLUSTERING ORDER BY (time DESC)
+    AND bloom_filter_fp_chance = 0.01
+    AND caching = {{'keys': 'ALL', 'rows_per_partition': 'ALL'}}
+    AND comment = ''
+    AND compaction = {{'class': 'SizeTieredCompactionStrategy'}}
+    AND compression = {{'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}}
+    AND crc_check_chance = 1.0
+    AND dclocal_read_repair_chance = 0.0
+    AND default_time_to_live = 0
+    AND gc_grace_seconds = 864000
+    AND max_index_interval = 2048
+    AND memtable_flush_period_in_ms = 0
+    AND min_index_interval = 128
+    AND read_repair_chance = 0.0
+    AND speculative_retry = '99.0PERCENTILE';",
+            TEST_GENERATION_TABLE
+        )
+    }
+
+    // Constructs mock table with the same schema as the original one's.
+    fn construct_stream_table_query() -> String {
+        format!(
+            "CREATE TABLE IF NOT EXISTS {} (
+    time timestamp,
+    range_end bigint,
+    streams frozen<set<blob>>,
+    PRIMARY KEY (time, range_end)
+) WITH CLUSTERING ORDER BY (range_end ASC)
+    AND bloom_filter_fp_chance = 0.01
+    AND caching = {{'keys': 'ALL', 'rows_per_partition': 'ALL'}}
+    AND comment = ''
+    AND compaction = {{'class': 'SizeTieredCompactionStrategy'}}
+    AND compression = {{'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}}
+    AND crc_check_chance = 1.0
+    AND dclocal_read_repair_chance = 0.0
+    AND default_time_to_live = 0
+    AND gc_grace_seconds = 864000
+    AND max_index_interval = 2048
+    AND memtable_flush_period_in_ms = 0
+    AND min_index_interval = 128
+    AND read_repair_chance = 0.0
+    AND speculative_retry = '99.0PERCENTILE';",
+            TEST_STREAM_TABLE
+        )
+    }
+
+    // Creates test keyspace and tables if they don't exist.
+    // Test data was sampled from a local copy of database.
+    async fn create_test_db(session: &Session) {
+        let mut query = Query::new(format!(
+            "CREATE KEYSPACE IF NOT EXISTS {} WITH replication
+                = {{'class':'SimpleStrategy', 'replication_factor': 3}};",
+            TEST_KEYSPACE
+        ));
+        query.set_consistency(Consistency::All);
+
+        session.query(query, &[]).await.unwrap();
+
+        // Create test tables containing information about generations and streams.
+        for query in vec![
+            construct_generation_table_query(),
+            construct_stream_table_query(),
+        ] {
+            session.query(query, &[]).await.unwrap();
+        }
+
+        // Delete all leftovers from previous tests.
+        for table in vec![
+            TEST_STREAM_TABLE.to_string(),
+            TEST_GENERATION_TABLE.to_string(),
+        ] {
+            session
+                .query(format!("TRUNCATE {};", table), &[])
+                .await
+                .unwrap();
+        }
+    }
+
+    // Populate test tables with given data.
+    async fn populate_test_db(session: &Session) {
```

Comment:
This function no longer works with a one node cluster because `.query` uses CL=QUORUM by default and the test keyspace now uses RF=3, so at least two nodes are needed for such query to work. I suggest extracting the `get_cluster_size` to a standalone function and use it both in `GenerationFetcher` and the tests.